assert/cmp/compare.go

```diff
@@ -0,0 +1,178 @@
+
+// Compare two complex values using https://godoc.org/github.com/google/go-cmp/cmp
+// and succeeds if the values are equal.
+func Compare(x, y interface{}, opts ...cmp.Option) func() (bool, string) {
+	return func() (bool, string) {
+		diff := cmp.Diff(x, y, opts...)
+		return diff == "", "\n" + diff
+	}
+}
+
+// Equal succeeds if x == y returns true
+func Equal(x, y interface{}) func() (success bool, message string) {
+	return func() (bool, string) {
+		return x == y, fmt.Sprintf("%v (%T) != %v (%T)", x, x, y, y)
+	}
+}
+
+// Len succeeds if the sequence has the expected length
+func Len(seq interface{}, expected int) func() (bool, string) {
+	return func() (success bool, message string) {
+		defer func() {
+			if e := recover(); e != nil {
+				success = false
+				message = fmt.Sprintf("type %T does not have a length", seq)
+			}
+		}()
+		value := reflect.ValueOf(seq)
+		if value.Len() == expected {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected %s to have length %d", seq, expected)
+	}
+}
+
+// NilError succeeds if the last argument is a nil error
+func NilError(args ...interface{}) func() (bool, string) {
+	return func() (bool, string) {
+		if len(args) == 0 {
+			return false, "NilError() requires one or more arguments"
+		}
+		return Nil(args[len(args)-1])()
+	}
+}
+
+// Contains succeeds if item is in collection. Collection may be a string, map,
+// slice, or array. reflect.DeepEqual() is used to compare item to each value in
+// collection.
+func Contains(collection interface{}, item interface{}) func() (bool, string) {
+	return func() (bool, string) {
+		colValue := reflect.ValueOf(collection)
+		if !colValue.IsValid() {
+			return false, fmt.Sprintf("nil does not contain items")
+		}
+		msg := fmt.Sprintf("%v does not contain %v", collection, item)
+
+		switch colValue.Type().Kind() {
+		case reflect.String:
+			itemValue := reflect.ValueOf(item)
+			success := strings.Contains(colValue.String(), itemValue.String())
+			return success, msg
+		case reflect.Map:
+			mapKeys := colValue.MapKeys()
+			for i := 0; i < len(mapKeys); i++ {
+				if reflect.DeepEqual(mapKeys[i].Interface(), item) {
+					return true, ""
+				}
+			}
+			return false, msg
+		case reflect.Slice, reflect.Array:
+			for i := 0; i < colValue.Len(); i++ {
+				if reflect.DeepEqual(colValue.Index(i).Interface(), item) {
+					return true, ""
+				}
+			}
+			return false, msg
+		default:
+			return false, fmt.Sprintf("type %T does not contain items", collection)
+		}
+	}
+}
+
+// Panics succeeds if f() panics
+func Panics(f func()) func() (bool, string) {
+	return func() (success bool, message string) {
+		defer func() {
+			if err := recover(); err != nil {
+				success = true
+			}
+		}()
+		f()
+		return false, "did not panic"
+	}
+}
+
+// EqualMultiLine succeeds if the two strings are equal. If they are not equal
+// the failure message will be the difference between the two strings.
+func EqualMultiLine(x, y string) func() (bool, string) {
```

Comment:
Should this be `EqualStrings()` or `EqualMultiLine()` ?

Reply:
🤔 `MultiLine` seems fine 👼